lib/calendar.rb

```diff
@@ -0,0 +1,32 @@
+module Calendar
+  def Calendar.date_range_include?(reservation, date)
```

Comment:
It's good to have these methods, but wouldn't it make sense to make them instance methods of the `Reservation` class or pulled into instance method of a `DateRange` class?  that way you could do?

```ruby
my_reservation.includes(date)

my_reservation.does_not_overlap(other_reservation)
```